lib/plugins/aws/invokeLocal/index.js

```diff
@@ -95,7 +96,20 @@ class AwsInvokeLocal {
 
     const callback = (err, result) => {
       if (err) {
-        throw err;
+        let errorResult;
+        if (err instanceof Error) {
+          errorResult = {
+            errorMessage: err.message,
+            errorType: err.constructor.name,
+          };
+        } else {
+          errorResult = {
+            errorMessage: err,
+          };
+        }
+
+        this.serverless.cli.consoleLog(chalk.red(JSON.stringify(errorResult, null, 4)));
+        process.exit(1);
```

Comment:
With this PR: https://github.com/serverless/serverless/pull/2836

You should not (have to) process.exit, and if you want to set the exit code, set process.exitCode or (preferably) reject the promise.